app/gui/src/controller/searcher/component/group.rs

```diff
@@ -93,14 +93,14 @@ impl Group {
 
     /// Create empty group referring to some module component.
     pub fn from_entry(component_id: component::Id, entry: &suggestion_database::Entry) -> Self {
-        let name: String = if entry.module.is_top_module() {
-            let project = &entry.module.project_name.project;
-            let module = entry.module.name();
+        let name = if entry.defined_in.is_top_element() || entry.defined_in.is_main_module() {
+            let project = &entry.defined_in.project().project;
+            let module = entry.defined_in.name();
             format!("{}.{}", project, module)
         } else {
-            entry.module.name().into()
+            entry.defined_in.name().to_owned()
         };
```

Comment:
Would it make sense to move this logic to a helper in `QualifiedName`?

Reply:
I don't think so. This code is very specific to the component groups, and a bit unusual.